rgw: allow authentication keystone with self signed certs

theanalyst opened the pull request:
This pr makes deployments that use keystone terminated with a self signed certificate to be used for authentication for radosgw, this is similiar to many openstack clients which allow for a `--insecure` switch to the option. Here based on the configurable `rgw keystone verify ssl` (which is true by default) we can allow for rgw to post requests with a keystone with self signed cert.

Fixes: #14853

yehudasa commented on src/rgw/rgw_http_client.h:
```diff
@@ -57,6 +58,10 @@ class RGWHTTPClient
     has_send_len = true;
   }
 
+  void set_insecure() {
```
@theanalyst change it to `set_insecure(bool flag)`

theanalyst replied:
sure will do

theanalyst replied:
Thinking of renaming  to set_verify_ssl(flag), so that config variable can directly be passed, I'll push the updated version..

yehudasa commented on src/rgw/rgw_http_client.h:
```diff
@@ -57,6 +58,10 @@ class RGWHTTPClient
   }
 
+  void set_insecure() {
+    insecure = true;
```
`insecure = flag;`

theanalyst replied:
sure

theanalyst commented:
I have an updated branch rebased on top of master.. https://github.com/ceph/ceph/compare/master...theanalyst:wip/rgw-insecure-keystone-rebase?expand=1 which worked before rebase, I couldn't yet test the build completely as master was broken, I'll update this pr as soon as I get a green build on master..

theanalyst commented:
@yehudasa I've rebased on top of current master (which is green now), as there was one more patch touching http_client, I've renamed the insecure option and added according to your comments. Let me know how this looks

yehudasa commented:
@theanalyst you can squash all these commits to one commit

theanalyst commented:
sure will do..

theanalyst commented:
@yehudasa rebased and squashed
